src/lib/yang/util.lua

```diff
@@ -128,6 +128,13 @@ function memoize(f, max_occupancy)
    end
 end
 
+-- XXX: lib/ctable.lua also defines set. Consider moving this function to core/lib.lua.
```

Comment:
For me, moving this function to `lib.yang.util` isn't a great intermediate step; I would move it to `core.lib` I think.